### PR TITLE
feat(Windows): Adds Windows support to react-native-video

### PR DESCRIPTION
**Pull request by rozele**
This PR adds react-native-windows support to react-native-video.  The Video component is implemented with a XAML MediaPlayerElement.  Most of the features implemented by Android (and some additional ones) are implemented by Windows.

Known issues and missing features include:
* onReadyForDisplay event
* local URI convention (e.g., "broadchurch" in examples changed to require("./broadchurch.mp4")
* `playableDuration` in `onVideoProgress` event is always 0.0
* `playInBackground` is not yet supported
* Volume settings are applied, but the UWP control does not handle it properly

### CODE REVIEW

File: example/index.windows.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  }
+
+  render() {
+    const flexCompleted = this.getCurrentTimePercentage() * 100;
+    const flexRemaining = (1 - this.getCurrentTimePercentage()) * 100;
+
+    return (
+      <View style={styles.container}>
+        <TouchableOpacity style={styles.fullScreen} onPress={() => {this.setState({paused: !this.state.paused})}}>
+          <Video source={require('./broadchurch.mp4')}
```
Comment (rozele) [inline example/index.windows.js:92]: > require('./broadchurch.mp4') [](start = 25, length = 28)

Note, the ability to just use `{uri: "broadchurch"}` is not supported on Windows. #WontFix

File: example/index.windows.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      </TouchableOpacity>
+    )
+  }
+
+  render() {
+    const flexCompleted = this.getCurrentTimePercentage() * 100;
+    const flexRemaining = (1 - this.getCurrentTimePercentage()) * 100;
+
+    return (
+      <View style={styles.container}>
```
Comment (rozele) [inline example/index.windows.js:90]: > style [](start = 12, length = 5)

This was copied from the index.android.js, consider looking at index.ios.js to see if there's any other features we can easily add. #Closed

**Comment by rozele** [comment]
> CodeFlow for GitHub does not download executable files. If you trust this file and would like to run it, please download it yourself through GitHub.

This should have been ignored.

---
Refers to: example/windows/VideoPlayer.VC.db:1 in e468fa9. [](commit_id = e468fa95b2089e9273f873a1847b089798dc493c, deletion_comment = False)

**Comment by rozele** [comment]
Be sure to update the README.

File: example/windows/VideoPlayer.sln [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+# Visual Studio 14
+VisualStudioVersion = 14.0.25420.1
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "VideoPlayer", "VideoPlayer\VideoPlayer.csproj", "{A027BE54-D118-49A4-8D84-0666A2A93E64}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ReactNative", "..\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj", "{C7673AD5-E3AA-468C-A5FD-FA38154E205C}"
+EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "ChakraBridge", "..\node_modules\react-native-windows\ReactWindows\ChakraBridge\ChakraBridge.vcxproj", "{4B72C796-16D5-4E3A-81C0-3E36F531E578}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ReactNativeVideo", "..\..\windows\ReactNativeVideo\ReactNativeVideo.csproj", "{E8F5F57F-757E-4237-AD23-F7A8755427CD}"
```
Comment (rozele) [inline example/windows/VideoPlayer.sln]: > "....\ [](start = 72, length = 7)

This should come from node_modules. #Closed

File: example/windows/VideoPlayer/VideoPlayer.csproj [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    <Content Include="Assets\Wide310x150Logo.scale-200.png" />
+  </ItemGroup>
+  <ItemGroup>
+    <ApplicationDefinition Include="App.xaml">
+      <Generator>MSBuild:Compile</Generator>
+      <SubType>Designer</SubType>
+    </ApplicationDefinition>
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj">
```
Comment (rozele) [inline example/windows/VideoPlayer/VideoPlayer.csproj:209]: > ="....\ [](start = 29, length = 8)

Double check this. #Closed

File: windows/ReactNativeVideo/ReactNativeVideo.csproj [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    <Compile Include="ReactVideoPackage.cs" />
+    <Compile Include="ReactVideoView.cs" />
+    <Compile Include="ReactVideoViewManager.cs" />
+    <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="ReactVideoEventType.cs" />
+    <Compile Include="ReactVideoEventTypeExtensions.cs" />
+    <EmbeddedResource Include="Properties\ReactNativeVideo.rd.xml" />
+  </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\example\node_modules\react-native-windows\ReactWindows\ReactNative\ReactNative.csproj">
```
Comment (rozele) [inline windows/ReactNativeVideo/ReactNativeVideo.csproj]: > ReactNative [](start = 104, length = 11)

Why isn't ReactNativeVideo referenced? #Closed

File: windows/ReactNativeVideo/ReactVideoView.cs [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+                }
+            }
+        }
+
+        public bool IsPaused
+        {
+            set
+            {
+                // TODO!
+                _isPaused = value;
```
Comment (rozele) [inline windows/ReactNativeVideo/ReactVideoView.cs:90]: Resolve issue with pause / resume. #Closed

File: windows/ReactNativeVideo/ReactVideoView.cs [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+                    MediaPlayer.SystemMediaTransportControls.IsEnabled = _isUserControlEnabled;
+                }
+            }
+        }
+
+        public double Volume
+        {
+            set
+            {
+                _volume = value;
```
Comment (rozele) [inline windows/ReactNativeVideo/ReactVideoView.cs:123]: > _volume [](start = 16, length = 7)

Double check volume is working. #Closed

File: windows/ReactNativeVideo/ReactVideoView.cs [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+                .EventDispatcher.DispatchEvent(
+                    new ReactVideoEvent(
+                        ReactVideoEventType.Seek.GetEventName(), 
+                        this.GetTag(),
+                        new JObject()));
+        }
+
+        private static async void RunOnDispatcher(DispatchedHandler action)
+        {
+            await CoreApplication.GetCurrentView().Dispatcher.RunAsync(CoreDispatcherPriority.Normal, action);
```
Comment (rozele) [inline windows/ReactNativeVideo/ReactVideoView.cs]: > CoreApplication [](start = 18, length = 15)

Double check. #Closed

File: README.md [hunk 9]
```diff
@@ -80,6 +80,62 @@ Under `.addPackage(new MainReactPackage())`:
...
+3. Select `node_modules\react-native-video\windows\ReactNativeVideo\ReactNativeVideo.csproj`
+
+**windows/myapp/myapp.csproj**
+
+Add a reference to `ReactNativeVideo` to your main application project.
+
+Using Visual Studio 2015:
+
+1. Right-click main application project > Add > Reference...
+2. Check `ReactNativeVideo` from Solution Projects.
```
Comment (GantMan) [inline README.md:100]: Best to give the user only visual studio version and not the XML.

Perhaps more of  Rightclick  project > Add to project > Check box

**Review by GantMan — APPROVED** [review]
LGTM 👍 

Needs a small readme touchup.

**Comment by kevinvangelder** [comment]
@mattapperson Do you have any feedback on this PR?

**Comment by mattapperson** [comment]
@kevinvangelder thanks for the ping. I really have no idea here... I know iOS and android but windows dev is not really something I know much about. As such I'm inclined to just merge and get support added and fix bugs as they come up...

**Pull request merged**